backend/app/rest/residents_routes.py

```diff
@@ -94,7 +93,10 @@ def delete_resident(resident_id):
         )
     except Exception as e:
         error_message = getattr(e, "message", None)
-        return jsonify({"error": (error_message if error_message else str(e))}), 500
+        if ("has existing log records" in str(e)):
+            return jsonify({"existing_log_record_error": "Resident has existing log records"}), 400
+        else:
+            return jsonify({"error": (error_message if error_message else str(e))}), 500
```

Comment:
nit: I feel like we could use a simpler string to throw the 400 here, but this part isn't necessary ; The main thing I wanted to point out was since this is an internal error (and the error string is effectively acting in the capacity of a variable), we might want to make it all capital, something like `ERROR_EXISTING_RECORD` or something